wordpress-mobile/WordPress-iOS
# [Stats] remove old code

Ref #10975
WPKit PR: https://github.com/wordpress-mobile/WordPressKit-iOS/pull/193
WPUI PR: https://github.com/wordpress-mobile/WordPressUI-iOS/pull/46


The main point if this PR is to remove the `WordpressComStatsiOS.framework` (which accounts for the massive change count). 

However, when I did so, there were several warnings about duplicate methods. Here's how I addressed them:

---
- warnings: 
  - instance method 'flashRowAtIndexPath:scrollPosition:flashLength:completion:' conflicts with same method from another category
  - instance method 'deselectSelectedRowWithAnimation:' conflicts with same method from another category
- Fix: Removed `UITableView+Extensions` from the Share extension, let it use the same methods from WordPressUI.

---
- warning: instance method 'trimNewlines' conflicts with same method from another category
- Fix: Removed `trimNewlines` from WPiOS, let it use the same method from WordPressKit.

---
- warnings: 
  - instance method 'initWithMinLineHeight:maxLineHeight:lineBreakMode:alignment:' conflicts with same method from another category
  - instance method 'initWithMinLineHeight:lineBreakMode:alignment:' conflicts with same method from another category
- Fix: Removed `NSMutableParagraphStyle+Helpers` from WPiOS, let it use the same methods from WordPressKit.

---
- warning: instance method 'setValueIfNotNil' conflicts with same method from another category
- Fix: Removed `NSMutableDictionary+Helpers` from WPiOS, let it use the same method from WordPressKit.

---
- warning: instance method 'md5' conflicts with same method from another category
- Fix: This method exists in WordPressUI and WordPressKit. I didn't want to remove functionality from either, so I renamed the WordPressUI method to `md5Hash`.

---
To test:
- Clean + build the app.
- Verify there are no errors or warnings regarding duplicate methods.
- Verify Stats works as expected.

---
Update release notes:
